src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -166,9 +173,11 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 
 # VM CREATE PARAMETER CONFIGURATION
 
-register_cli_argument('vm create', 'name', name_arg_type, validator=_resource_not_exists('Microsoft.Compute/virtualMachines'))
+register_cli_argument('vm create', 'new_vm_name', options_list=('--name', '-n'), metavar='NAME',
```

Comment:
So that this arg name would not pick the default from config, and we only need to do it for `create`, It appears the arg matching we have is based on string match. I can special case in the core layer to skip any `create` command like we did in other places, but again since it is limited on create command, so I do it at command module layer